Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs

```diff
@@ -0,0 +1,49 @@
+using Content.Shared.DeltaV.Shuttles.Systems;
+using Content.Shared.Tag;
+using Content.Shared.Whitelist;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.DeltaV.Shuttles.Components;
+
+/// <summary>
+/// A shuttle console that can only ftl-dock between 2 grids.
+/// The shuttle used must have <see cref="DockingShuttleComponent"/>.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedDockingConsoleSystem))]
+[AutoGenerateComponentState]
+public sealed partial class DockingConsoleComponent : Component
+{
+    /// <summary>
+    /// Title of the window to use
+    /// </summary>
+    [DataField(required: true)]
+    public LocId WindowTitle;
+
+    /// <summary>
+    /// Airlock tag that it will prioritize docking to.
+    /// </summary>
+    [DataField(required: true)]
+    public ProtoId<TagPrototype> DockTag;
+
+    /// <summary>
+    /// A whitelist the shuttle has to match to be piloted.
+    /// </summary>
+    [DataField(required: true)]
+    public EntityWhitelist ShuttleWhitelist = new();
+
+    /// <summary>
+    /// The shuttle that matches <see cref="ShuttleWhitelist"/>.
+    /// If this is null a shuttle was not found and this console does nothing.
+    /// </summary>
+    [DataField]
+    public EntityUid? Shuttle;
+
+    /// <summary>
+    /// Whether <see cref="Shuttle"/> is set on the server or not.
+    /// Client can't use Shuttle outside of PVS range so that isn't networked.
+    /// </summary>
+    [DataField, AutoNetworkedField]
```

Comment:
why does this need to be networked if it's never being used on the client

Reply:
meant to have a thing that displays an error when theres no shuttle :trollface: